pkg/prometheus/server/operator.go

```diff
@@ -521,7 +530,7 @@ func (c *Operator) Run(ctx context.Context) error {
 
 	c.addHandlers()
 
-	if c.kubeletSyncEnabled {
+	if c.kubeletSyncEnabled && !operator.IsSingleNamespace(c.config.Namespaces) {
```

Comment:
why do we need this?

Reply:
level=error ts=2023-09-01T11:39:50.560491434Z caller=operator.go:626 component=prometheusoperator msg="Syncing nodes into Endpoints object failed" err="listing nodes failed: nodes is forbidden: User \"system:serviceaccount:rio:prometheus-operator\" cannot list resource \"nodes\" in API group \"\" at the cluster scope"
That's the error log if disable this limit.